README.md

```diff
@@ -1 +1,45 @@
-# A general purpose, high performance write-ahead-log
\ No newline at end of file
+# A general purpose, high performance write-ahead-log
+
+## Usage
+
+First the wal needs to be opened by calling `New(string path)`. This will create a new WAL at the provided path or load
+an existing one. The latter will recover the WAL and return all the updates of transactions that were not completed.
+
+```
+// Open the WAL.                                                                                 
+updates, wal, err := New(walPath)                                                            
+if err != nil {                                                                                  
+    return nil, err                                                                              
+} 
+```
+
+The wal can then be used to create a `Transaction` like this using a set of updates:
+
+```
+// Create the WAL transaction.                                                                   
+tx, err := ca.wal.NewTransaction(updates)                                                        
+if err != nil {                                                                                  
+    return err                                                                                   
+}   
+```
+
+An `Update` consists of a Name, Version and Instructions. One transaction can hold multiple updates.
+After the `Transaction` is created the caller might want to do some kind of setup. Once completed the next step is to signal the WAL that the setup is complete.
+
+```
+// Signal completed setup and then wait for the commitment to finish.                            
+errChan := tx.SignalSetupComplete()                                                              
+err = <-errChan                                                                                  
```

Comment:
I would put a comment here emphasizing that the wal is going to do an fsync before returning.